### DISCUSSION TITLE
Keep-alive Connection

### DISCUSSION THREAD

**Discussion by scholes231** [Havret/dotnet-activemq-artemis-client#481] [General]
In my Asp.net project a create a Consumer like this:
builder.Services.TryAddScoped<TConsumer>();
builder.AddConsumer(address, routingType, queueName, HandleMessage<TMessage, TConsumer>);

And we noticed that at certain intervals (1, 3, or more minutes later), there is a reconnect occurring. In the logs: ClosedByPeer=True, Error=error(condition:amqp:connection:forced,description:The transport 'TcpTransport' is closed.)
What do I wrong? How can I set keep-alive connection

**Comment by Havret** [comment]
Hi @scholes231,

Setting keep-alive is not possible with the current version ([2.15.0](https://www.nuget.org/packages/ArtemisNetClient/2.15.0)). But with the latest 2.16.0 you can set keep alive as follows:

```csharp
var factory = new ConnectionFactory();
factory.TCP.KeepAliveTime = 1000 * 30; // 30 seconds
factory.TCP.KeepAliveInterval = 1000; // 1 seconds 
```

Please let me know if this solves your issue.

Best regards,
Havret
